nomad/state/state_store_task_group_volume_claims.go

```diff
@@ -106,20 +116,36 @@ func (s *StateStore) GetTaskGroupHostVolumeClaimsForTaskGroup(ws memdb.WatchSet,
 	}
 	ws.Add(iter.WatchCh())
 
-	// Filter out by ns, jobID and tg
 	filter := memdb.NewFilterIterator(iter, func(raw interface{}) bool {
 		claim, ok := raw.(*structs.TaskGroupHostVolumeClaim)
 		if !ok {
+			return false
```

Comment:
```suggestion
			return true
```

I hate the API of "filter" because this is so easy to mix up. :grinning:  Of course, if we had the v2/generics version of immutable radix in memdb we wouldn't even have this code, but that's going to be a cross-team lift to make happen (and benchmark).

Reply:
oof, thanks. And yes this is very confusing :(